Formula/diskonaut.rb

```diff
@@ -0,0 +1,18 @@
+class Diskonaut < Formula
+  desc "Terminal visual disk space navigator"
+  homepage "https://github.com/imsnif/diskonaut"
+  url "https://github.com/imsnif/diskonaut/archive/0.9.0.tar.gz"
+  sha256 "93564a195a62796f95eacc154ad10df8f1050c8a0b2e2e22a0612228c930c1f5"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    output = shell_output "#{bin}/diskonaut --version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hey @SMillerDev - thanks for the quick response!

Diskonaut is a terminal UI, so it might be a little difficult to test in this manner. It changes the terminal to "raw mode" and so asserting anything on the stdout will be a little more involved (it would depend on the cols/rows of the testing terminal).

I took inspiration from a different crate I maintain and its brew formula: https://github.com/Homebrew/homebrew-core/pull/48504/files

An equivalent test I can do in the case of Diskonaut would be to run it and then get this error: `<"Error: Failed to get stdout: are you trying to pipe 'diskonaut'?">.` Would that be better?

Reply:
Another option: With `bandwhich` (the linked crate), the test provides it an invalid argument and then checks for an error message.
With some trivial code change in `diskonaut` (this crate), I can validate the arguments and only then request the STDOUT. So then we can replicate that test here in this formula (eg. run `diskonaut /non/existent/path` and expect a proper error).
I'll have to release a new version and then update this PR, but it's no biggie. Is this preferable?

Reply:
I think the pipe failure is good enough.

Reply:
Cool - done. :+1: